src/class-main.php

```diff
@@ -115,4 +151,68 @@ protected function get_service_classes() : array {
       View\Post_View_Filter::class,
     );
   }
+
+  /**
+   * Array of service classes used in tests.
+   *
+   * @return array
+   */
+  protected function test_service_classes(): array {
+    return [
+      // Config.
+      Config::class,
+
+      // Authorization.
+      Integrations\Authorization\HMAC::class,
+
+      // HTTP.
+      Integrations\Core\Guzzle_Client::class => array(
+        Client::class,
+      ),
+
+      // Base route.
+      Rest\Test_Route::class => array(
+        Config::class,
+        Integrations\Authorization\HMAC::class,
+        Captcha\Basic_Captcha::class,
+      ),
+
+      // Email route.
+      Rest\Send_Email_Route::class => array(
+        Config::class,
+        Captcha\Basic_Captcha::class,
+      ),
+
+      // Buckaroo routes.
+      Integrations\Buckaroo\Buckaroo::class => array(
+        Integrations\Core\Guzzle_Client::class,
+      ),
+      Rest\Buckaroo_Response_Handler_Route::class => array(
+        Config::class,
+        Integrations\Buckaroo\Buckaroo::class,
+        Integrations\Authorization\HMAC::class,
+      ),
+    ];
+  }
+
+  /**
+   * Get the list of services to register.
+   *
+   * A list of classes which contain hooks.
+   *
+   * @return array<string> Array of fully qualified class names.
+   */
+  protected function get_service_classes() : array {
+    return $this->is_test ? $this->test_service_classes() : $this->prod_service_classes();
+  }
+
+  /**
+   * Provides additional / different services depending on if we're in test or not.
+   *
+   * @param  boolean $is_test Set to true if running tests.
+   * @return void
+   */
+  public function set_test( bool $is_test ): void {
```

Comment:
This is interesting. So during the test, you need to first set this to `true`, then you can bootstrap the plugin? A bit odd approach, but if it works 🤷‍♂️

Reply:
Yup, I use a base class for tests which (if it needs injection) does this (in order to get the container for tests). For example:

```php
abstract class BaseRouteTest extends BaseTest {

  protected function _inject(Main $main, HMAC $hmac)
  {
    $main->set_test(true);
    $this->di_container = $main->build_di_container();
    ...

```